addon/utils/load-relationship.js

```diff
@@ -7,6 +7,7 @@ export default function loadAll(model, relationship, dest, options = {}) {
         page: page
     };
     query = Ember.merge(query, options || {});
+    Ember.set(model, 'query-params', query);
```

Comment:
I agree that this could potentially be problematic.  I am building the URL myself in findHasMany to embed both `parent` and `root`, but I still needed to pass on any other query params when I build the url.  I put them on the model so I would have them in findHasMany.  Maybe there is a better way to pass along this information.